src/resources/material.js

```diff
@@ -169,24 +42,21 @@ Object.assign(pc, function () {
         },
 
         open: function (url, data) {
-            var material = new pc.StandardMaterial();
-
-            // TODO: this is a bit of a mess,
-            // Probably should create a new data block for the material
-            // and put it on the asset. This preserves originally loaded asset data
-            // and can be removed/cleared when asset is unloaded.
-            if (!data.parameters) {
-                this._createParameters(data);
+            var material = this._parser.parse(data);
+
+            // temp storage for engine-only as we need this during patching
+            if (data._engine) {
```

Comment:
When you say `temp storage` does this mean you're changing this?

Reply:
This basically how it worked before, but I changed the names slightly to be a bit more explicit.

If you load a material from a json file (i.e. engine-only) we add this _engine field to the data when it is loaded (we know we are engine on when material is requested by a URL)

Then in the open() method we copy the data loaded onto the material resource temporarily.

Finally in patch() we can assign the data from the material resource into the asset.